### ISSUE TITLE
Iterable type hint makes it easy to pass in wrong type of key argument

### ISSUE THREAD

**Issue by mikeyk** [alisaifee/coredis#48]
First of all, thank you so much for stewarding this library! The latest versions look great.

One notable difference in moving from `aioredis` is the behavior of functions that can either take one or many keys, for example `exists`. I think the approach `coredis` takes is reasonable -- which is to require that a list be passed in, even if it's a single-element list. However, the current type hinting makes it possible to accidentally pass in a single key instead, like:

```
await redis.exists("mykey")
```

Which will destructure into testing for each character in "mykey".

Would you be open to tightening the `Iterable` typehint on those commands? I'm happy to put up a PR if so. I was thinking something like `Union[List, Tuple, Set, ValuesView, ItemsView]` which should cover the majority of use cases for passing in iterables-but-not-strings into those functions.

**Comment by alisaifee** [comment]
> First of all, thank you so much for stewarding this library! The latest versions look great.

Thank you for giving it a chance and the feedback 🙇 

The issue with switching out variadic arguments when they are not optional is I think the most "annoying" decision I've made when adopting / migrating `aredis` to `coredis`. To be honest it was done purely out of laziness because I wanted to be able to rely on the redis command spec for optionality and be able to generate a reasonably consistent set of signature mappings for each redis command (i.e. let a script generate the methods for me). At the time I completely forgot about how strings and bytes are tacked onto `Iterable` and basically mess everything up both for runtime and static checking. 

I'd considered two options:
1) Describe a Protocol that allows everything but strings and bytes
2) throw in the towel and add an override for common methods which face this issue (`delete` & `exists`  are probably the most obvious ones)

Now that I think about it pinning it to `Union[List, Tuple, Set, ValuesView, ItemsView]` really does look perfectly reasonable. The only (aesthetic) issue I have with that is how verbose the api docs become with sphinx, but I'm sure there's something that can be figured out with a TypeAlias. 

**TL;DR** yes I think we should go ahead with your proposal. I'm getting close to completing a rather large refactor of the guts to make the type safety a bit more serious than just on the surface of the API, but if you have the bandwidth to put up a PR I'd be happy to push out a dot release. I think adding a new type alias in [typing.py](../tree/master/coredis/typing.py#L78) and replacing `keys: Iterable[KeyT]` with `keys: Keys` (and maybe a `Values`) (I suck at naming) would work?

**Comment by alisaifee** [comment]
Just a heads up, I managed to get something working on master (unreleased). This will **probably** be going out in `3.5`. I'll leave this open until I do cut a release and in case you have some feedback.

The TL;DR is that I did end up going with a union along the lines of what you had suggested - details [here](https://coredis.readthedocs.io/en/latest/api.html#input-types).

On a side note, thank you for bringing this up and the suggestion. Turns out a handful of tests were infact calling `delete("fubar")` instead of `delete(["fubar"])` etc and had been accidentally passing. Since the CI runs with runtime checking they started failing once I switched `Iterable` with `Parameters`. Here's an [example run](https://github.com/alisaifee/coredis/actions/runs/2284628832) if you're interested.

**Comment by alisaifee** [comment]
This is now addressed in [3.5.0](https://github.com/alisaifee/coredis/releases/tag/3.5.0)
